src/__tests__/useMultiStateValidator.ts

```diff
@@ -0,0 +1,125 @@
+import { act, renderHook, RenderHookResult } from '@testing-library/react-hooks';
+import { useState } from 'react';
+import { MultiStateValidator, useMultiStateValidator } from '../useMultiStateValidator';
+import { UseValidatorReturn, ValidityState } from '../useStateValidator';
+
+interface Mock extends jest.Mock {}
+
+describe('useMultiStateValidator', () => {
+  it('should be defined', () => {
+    expect(useMultiStateValidator).toBeDefined();
+  });
+
+  const defaultStatesValidator = (states: number[]) => [states.every(num => !(num % 2))];
+
+  function getHook(
+    fn: MultiStateValidator<any, number[]> = jest.fn(defaultStatesValidator),
+    initialStates = [1, 2],
+    initialValidity = [false]
+  ): [MultiStateValidator<any, number[]>, RenderHookResult<any, [Function, UseValidatorReturn<ValidityState>]>] {
+    return [
+      fn,
+      renderHook(
+        ({ initStates, validator, initValidity }) => {
+          const [states, setStates] = useState(initStates);
+
+          return [setStates, useMultiStateValidator(states, validator, initValidity)];
+        },
+        {
+          initialProps: {
+            initStates: initialStates,
+            initValidity: initialValidity,
+            validator: fn,
+          },
+        }
+      ),
+    ];
+  }
+
+  it('should return an array of two elements', () => {
+    const [, hook] = getHook();
+    const res = hook.result.current[1];
+
+    expect(Array.isArray(res)).toBe(true);
+    expect(res.length).toBe(2);
+  });
+
+  it('should call validator on init', () => {
+    const [spy] = getHook();
+
+    expect(spy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should call validator on any of states changed', () => {
+    const [spy, hook] = getHook();
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    act(() => hook.result.current[0]([4, 2]));
+    expect(spy).toHaveBeenCalledTimes(2);
+  });
+
+  it("should NOT call validator on it's change", () => {
+    const [spy, hook] = getHook();
+    const newValidator: MultiStateValidator<any, number[]> = jest.fn(states => [states!.every(num => !!(num % 2))]);
+
+    expect(spy).toHaveBeenCalledTimes(1);
+    hook.rerender({ validator: newValidator });
+    expect(spy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should throw if states is not an object', () => {
+    try {
+      // @ts-ignore
+      getHook(defaultStatesValidator, 123);
+    } catch (err) {
+      expect(err).toBeDefined();
+      expect(err instanceof Error).toBe(true);
+      expect(err.message).toBe('states expected to be an object or array, got number');
```

Comment:
Better to use [toThrow](https://jestjs.io/docs/en/expect#tothrowerror) in this case, as this test could give a false negative at the moment if no error is thrown (and it's also the proper matcher for errors).

Reply:
Dont understand how it can give false negative >_>

Reply:
The method called doesn't throw any error --> there is no error to catch --> all the expects inside the catch are not called: the test pass (no expect called so far) but it shouldn't. But also, `toThrow` is proper matcher to call here, much simpler, readable and maintainable.

Reply:
Ah, yes, this one will cause false-negative.
But it does not work for the hook somewhy. If to do: 
```typescript
expect(() => {
      // @ts-ignore
      getHook(defaultStatesValidator, 123);
    }).toThrowError('states expected to be an object or array, got number');
```
it will fail with 
```
Error: expect(received).toThrowError(expected)

Expected substring: "states expected to be an object or array, got number"

Received function did not throw

Reply:
Somewhy, error been swallowed inside the `hook.result.error` and i have to rethrow it =\